tests/TestSlideData.cpp

```diff
@@ -0,0 +1,26 @@
+#include "catch.hpp"
+using namespace Catch;
+
+#include "SightLib.hpp"
+
+TEST_CASE("file not exists", "[SlideData]")
+{
+  sight::SlideData data;
+  REQUIRE_THROWS_WITH(data.open("does_not_exist"), Contains("Could not find file"));
+}
+
+TEST_CASE("file invalid format", "[SlideData]")
+{
+  sight::SlideData data;
+  REQUIRE_THROWS_WITH(
+      data.open("../CMakeLists.txt"), Contains("Could not detect filetype vendor"));
+}
+
+TEST_CASE("open test file", "[SlideData]")
+{
+  // TODO: don't want to include this 200MB file in the repo....
```

Comment:
We probably do want to include it, if the test is useful. 200MB is not so big.

Reply:
Joshua is going to get back to us with a smaller file. The 200MB one is just one that I downloaded from the openslide web-page.